pilot/pkg/serviceregistry/serviceentry/controller.go

```diff
@@ -777,15 +759,15 @@ func (s *Controller) doEdsUpdate(keys sets.Set[instancesKey]) {
 
 // buildEndpoints builds endpoints for the instance keys.
 func (s *Controller) buildEndpoints(keys map[instancesKey]struct{}) map[instancesKey][]*model.IstioEndpoint {
-	var endpoints map[instancesKey][]*model.IstioEndpoint
-	allInstances := []*model.ServiceInstance{}
+	allInstances := make([]*model.ServiceInstance, 0, len(keys))
```

Comment:
This is not correct, below i can be not only one instance

Reply:
reverted